quesma/ingest/processor.go

```diff
@@ -560,14 +564,30 @@ func (ip *IngestProcessor) processInsertQuery(ctx context.Context,
 		}
 		processed = append(processed, result)
 	}
+	jsonData = processed
+
+	encodings := make(map[schema.FieldEncodingKey]string)
+	// we are doing two passes, e.g. calling transformFieldName twice
+	// first time we populate encodings map
+	// second time we do field encoding
+	// This is because we need to know all field encodings before we start
+	// transforming fields, otherwise we would mutate json and populating encodings
+	// which would introduce side effects
```

Comment:
yea that's a bummer, but I don't have any better idea for now. I think it might hit us faster than we might expect though, esp. that for clickhouse ingest its advices to have significantly larger bulk payloads than elasticsearch (hundreds of thousands vs few thousands)

Reply:
true, but this can be further optimized. One of the method would be to rewrite `transformFieldName` function, but I think it's too much for this PR